R/snapshot.R

```diff
@@ -986,15 +989,29 @@ renv_snapshot_filter_report_missing <- function(missing, type) {
     else
       "Use `renv::dependencies()` to see where this package is used in your project."
   )
-  renv_pretty_print(
-    values = csort(unique(missing)),
+  renv_pretty_bullets(
+    values = unique(missing),
     preamble = preamble,
     postamble = postamble
   )
 
-  cancel_if(interactive() && !proceed())
-  TRUE
+  choices <- c(
+    "Install the packages, then snapshot",
+    "Snapshot, just using the currently installed packages",
+    "Cancel"
+  )
+  choice <- tryCatch(
+    menu(choices, graphics = FALSE, title = "What do you want to do?"),
+    interrupt = function(cnd) 0
+  )
```

Comment:
This also needs a bit of work to make it testable, and it'll be worth also using where `utils::select.list()` currently is.

Reply:
What are the pros and cons of `menu()` vs `select.list()`?

Reply:
Lets discuss in #1372, but in brief, I find the returned value to be easier to work with.

Reply:
👍